### ISSUE TITLE
Always asking "close and save"

### ISSUE THREAD

**Issue by frazzmark** [dschuermann/document-viewer#64]
When closing a PDF opened from downloads,  it always asks to save the document, even though it has already been downloaded and saved by the browser.
Opened from "Download" app in the drawer.

**Comment by kiike** [comment]
This bug is still present on 2.6 from f-droid.org. It is triggered when opening a PDF file from an external program, such as `Downloads` or `Lightning Browser`, but not when opening PDFs from the List view from the application.

**Comment by tuxor1337** [comment]
Yes, it's still present in 2.6 (has anybody stated otherwise?). Would you please tell me what's your preferred behaviour? Would you like document viewer to _never_ state this dialog? But what if your file is indeed only a temporary file? Wouldn't you want an option to save the file before closing and losing the data?

If you want to look into this, maybe a look into the following lines will help: https://github.com/dschuermann/document-viewer/blob/master/document-viewer/src/main/java/org/ebookdroid/ui/viewer/ViewerActivityController.java#L769

There it is checked whether the property `temporary` of the current `ContentScheme` is set to true and only in this case the SaveAndClose action is triggered. If you look into ContentScheme.java (https://github.com/dschuermann/document-viewer/blob/master/document-viewer/src/main/java/org/emdev/common/content/ContentScheme.java), you see that this is the case exactly for email attachments and SMP/HTTP/HTTPS sources.

**Comment by frazzmark** [comment]
The problem is still there when I open the file from "Downloads" app, but when open it from Firefox and then close it, everything seems fine.

**Comment by frazzmark** [comment]
Confirmed in 2.7.3, PDF opened from Downloads app.

**Comment by ericwa** [comment]
I looked into why this happens with the "Downloads" app. When you tap on a file in the "Downloads" app, it opens with the content:// URI scheme (as a guess, maybe so apps can open downloads without reading from the filesystem?), which document-viewer considers as "temporary".

For comparison, I checked Adobe Acrobat - if it downloads a document from an HTTP source, it  automatically saves to /sdcard/Download/Adobe Acrobat. I'm not sure if I like that, because it can clutter your downloads folder. It doesn't save an extra copy of documents opened from the "Downloads" app though, which is good.

I suggest we don't show the "Save and close" dialog for content:// URI's - the content:// URI means another application controls the document, so I guess it's the other app's responsibility to offer a UI for saving the document, e.g. in the case of an email attachment.

**Comment by tuxor1337** [comment]
Okay, that sounds reasonable. I tried to find the exact meaning of the scheme "content" on the web, but it doesn't seem to be totally well-defined. In any case, there's no reason to assume that all objects that are accessed via a "content" scheme are temporary.

Do you think it's safe to remove the "key" property from the definition of CONTENT in https://github.com/SufficientlySecure/document-viewer/blob/master/document-viewer/src/main/java/org/emdev/common/content/ContentScheme.java#L26 ?

**Comment by ericwa** [comment]
Maybe it's clearer to just add an extra parameter to the constructors, since the way "temporary" is set currently is sort of confusing? I made a PR https://github.com/SufficientlySecure/document-viewer/pull/140
